v1/linkml-schemas/lifeEvents.yaml

```diff
@@ -0,0 +1,99 @@
+id: https://vocab.account.gov.uk/linkml/life-events-schema
+name: lifeEvents
+version: 1.0
+description: >-
+  Vocabulary for life events
+
+prefixes:
+  linkml: https://w3id.org/linkml/
+  vc: https://www.w3.org/2018/credentials/v1/
+  di_vocab: https://vocab.account.gov.uk/v1/
+imports:
+  - linkml:types
+  - ./events
+  - ./person
+default_curi_maps:
+  - semweb_context
+default_prefix: di_vocab
+default_range: string
+
+classes:
+  DeathRegistrationJWTClass:
+    is_a: SETClass
+    attributes:
+      events:
+        range: DeathRegistrationEventMappingClass
+        required: true
+  DeathRegistrationEventMappingClass:
+    attributes:
+      "deathRegistration":
+        range: DeathRegistrationEventClass
+        required: true
+        alias: "https://ssf.account.gov.uk/v1/deathRegistration"
+  DeathRegistrationUpdateJWTClass:
+    is_a: SETClass
+    attributes:
+      events:
+        range: DeathRegistrationUpdateEventMappingClass
+        required: true
+  DeathRegistrationUpdateEventMappingClass:
+    attributes:
+      "deathRegistrationUpdate":
+        range: DeathRegistrationUpdateEventClass
+        required: true
+        alias: "https://ssf.account.gov.uk/v1/deathRegistrationUpdate"
+  DeathRegistrationEventClass:
+    attributes:
+      subject:
+        range: DeathRegistrationSubjectClass
+    slots:
+      - deathRegistrationID
+      - deathDate
+      - freeFormatDeathDate
+      - deathRegistrationTime
+  DeathRegistrationUpdateEventClass:
+    attributes:
+      subject:
+        range: DeathRegistrationSubjectClass
+    slots:
+      - deathRegistrationID
+      - deathDate
+      - freeFormatDeathDate
+      - recordUpdateTime
+      - deathRegistrationUpdateReason
+  DeathRegistrationSubjectClass:
+    is_a: PersonExtendedMatchingClass
+    slot_usage:
+      birthDate:
+        range: ISODateClass
```

Comment:
We have "DeceasedBirthDate" and friends in the [schema](https://docs.google.com/document/d/1e6QbX8pOCBbuIywTXI5cMXl_U4yYljVk/edit) - and generally birth date is exceedingly useful for matching. If it turns out we don't get it then I think that would be a problem that HMPO need to fix?

Reply:
I think it's specifically that the "description" isn't set on the birthDate (there's no "reason it's not a full date" field in the schema. I think we can just not set a value for it in our mapping, and leave the possibility that it has a description value in the schema.

Reply:
Ah I see, sorry. This is because we're using ISODateClass in two cases because both deathDate and birthDate might be partial.

I guess we could make a new class that excludes `description` but I think it doesn't do much harm leaving it in the schema even if it's unused currently in that context.

Reply:
Happy with that, no harm in keeping it and setting it to null :)

Reply:
Can confirm the schema validation won't allow `null` - it will allow empty string, but don't do that either :-)

Reply:
57604 removes `description` from `birthDate` on the example files.